### PR TITLE
Badge campaigns

### PR DESCRIPTION
**Pull request by michae1T**

### CODE REVIEW

File: core/src/main/scala/com/gilt/urbanairship/Campaigns.scala [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package com.gilt.urbanairship
+
+case class Campaigns(categories : Seq[String])
```
Comment (fiadliel) [inline core/src/main/scala/com/gilt/urbanairship/Campaigns.scala]: I would generally prefer concrete types in case classes (usually in this case, `List`).

**Comment by fiadliel** [comment]
Because it creates guarantees around behaviour and performance. For example, Seq could be an infinite Stream. Also, pattern matching can be guaranteed to work with `::`, where `Seq` could be anything. And this also makes explicit that this is a list, where `Seq.apply` will create lists anyway (so the runtime value is the same as when code used that constructor).

**Comment by fiadliel** [comment]
Also see https://gist.github.com/djspiewak/2ae2570c8856037a7738

**Comment by michae1T** [comment]
the infinite stream point in particular gives me pause in a theoretical sense.  although I'll have to put more thought into the practical implications of walking the other way on a larger scale.  Seq is endemic.

**Comment by michae1T** [comment]
I am not able to merge or publish to maven central... think you can publish still right?

**Pull request merged**